### PR TITLE
move all DSOs to large pages in LD_PRELOAD lib and also fix a few issues

### PR DESCRIPTION
**Pull request by rdementi**

### CODE REVIEW

File: large_page-c/large_page.c [hunk 1]
```diff
@@ -287,11 +287,11 @@ static void AlignRegionToPageBoundary(mem_range* r) {
 }
 
 static map_status CheckMemRange(mem_range* r) {
-  if (r->from == NULL || r->to == NULL || r->from > r->to) {
+  if (r->from == NULL || r->to == NULL) {
     return map_invalid_region_address;
   }
 
-  if (r->to - r->from < HPS) {
+  if (r->to - r->from < HPS || r->from > r->to) {
```
Comment (suresh-srinivas) [inline large_page-c/large_page.c:294]: Does this condition happen where from > to? We were treating it as an invalid region

**Review by suresh-srinivas — APPROVED** [review]
Thanks for the contribution. This Looks good to me. Is it possible to put the printfs under debug and use stderr? I don't know how it would work when used under systemd

File: large_page-c/large_page.c (hunk 1 above)
Comment (rdementi) [reply large_page-c/large_page.c:294]: "r->from > r->to" condition occurs for smaller regions after AlignRegionToPageBoundary is called ('from' is aligned up and 'to' is aligned down). This part of commit is not critical. I can drop it if needed.

Comment (rdementi) [reply large_page-c/large_page.c:294]: before AlignRegionToPageBoundary : from < to

File: large_page-c/lp_preload.c [hunk 2]
```diff
@@ -1,7 +1,49 @@
...
+#include <link.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <regex.h>
 #include "large_page.h"
 
+void printErr (map_status status, const char * lib)
+{
```
Comment (uttampawar) [inline large_page-c/lp_preload.c:11]: nipick. Please put open curly braces on the same line as function definition matching with rest of the coding standard.

File: large_page-c/lp_preload.c [hunk 3]
```diff
@@ -10,7 +52,23 @@ void __attribute__((constructor)) map_to_large_pages() {
...
-  if (status != map_ok) goto fail;
+  if (status != map_ok)
+  {
+    printErr(status, "static code");
+  }
+
+  regex_t ignoreReg;
+  const char * ignoreStr = secure_getenv("LP_IGNORE");;
+  if (ignoreStr == NULL || regcomp(&ignoreReg, ignoreStr, REG_EXTENDED) != 0)
+  {
```
Comment (uttampawar) [inline large_page-c/lp_preload.c:63]: nitpick. Same as before about the curly braces.

**Review by uttampawar — APPROVED** [review]
LGTM.

**Comment by rdementi** [comment]
@uttampawar @suresh-srinivas  Thanks a lot for your comments. I think I have addressed your comments in the latest commit I have pushed. Please review

**Review by uttampawar — APPROVED** [review]
LGTM

File: large_page-c/large_page.c (hunk 1 above)
Comment (suresh-srinivas) [reply large_page-c/large_page.c:294]: Thanks, that makes sense.

**Pull request merged**